src/main/java/com/redhat/devtools/intellij/lsp4ij/LSPIJUtils.java

```diff
@@ -116,40 +114,90 @@ public static URI toUri(File file) {
         }
     }
 
-    public static URI toUri(PsiFile file) {
-        return toUri(file.getVirtualFile());
+
+    public static @NotNull URI toUri(@NotNull Module module) {
+        VirtualFile[] roots = ModuleRootManager.getInstance(module).getContentRoots();
+        if (roots.length > 0) {
+            return toUri(roots[0]);
```

Comment:
this is not enough. The generated-sources content root shows up 1st in a myproject.main module